File: contest/q1.py
from collections import deque

def find_minimum_clicks(N, links, start_page, end_page):
    # Create a graph as an adjacency list
    graph = {}
    for i in range(1, N + 1):
        graph[i] = links[i - 1]

    # Initialize the BFS queue and visited set
    queue = deque([(start_page, 0)])  # Each element is a tuple (page, clicks)
    visited = set()

    while queue:
        current_page, num_clicks = queue.popleft()

        if current_page == end_page:
            return num_clicks

        visited.add(current_page)

        for neighbor in graph[current_page]:
            if neighbor not in visited:
                queue.append((neighbor, num_clicks + 1))

    return -1  # If end page cannot be reached

# Input
N = int(input())
links = [list(map(int, input().split())) for _ in range(N)]
start_page, end_page = map(int, input().split())

# Find and print the minimum clicks
result = find_minimum_clicks(N, links, start_page, end_page)
print(result)

File: contest/q1.cpp
#include <bits/stdc++.h>

using namespace std;

int findMinimumClicks(unordered_map<int, vector<int>>& webPages, int startPage, int endPage) {
    if (startPage == endPage) {
        return 0;
    }

    queue<int> q;
    unordered_map<int, int> distance;

    q.push(startPage);
    distance[startPage] = 0;

    while (!q.empty()) {
        int currentPage = q.front();
        q.pop();

        for (int linkedPage : webPages[currentPage]) {
            if (linkedPage == endPage) {
                return distance[currentPage] + 1;
            }
            if (distance.find(linkedPage) == distance.end()) {
                distance[linkedPage] = distance[currentPage] + 1;
                q.push(linkedPage);
            }
        }
    }

    return -1; // End page cannot be reached
}

int main() {
    int N;
    cin >> N;
    cin.ignore(); // Consume the newline character

    unordered_map<int, vector<int>> webPages;

    for (int i = 1; i <= N; i++) {
        string line;
        getline(cin, line);
        istringstream iss(line);
        int linkedPage;

        while (iss >> linkedPage) {
            webPages[i].push_back(linkedPage);
        }
    }

    int startPage, endPage;
    cin >> startPage >> endPage;

    int minClicks = findMinimumClicks(webPages, startPage, endPage);

    cout << minClicks ;

    return 0;
}